### ISSUE TITLE
Custom Account username/password are invisibly retained after switching to a BUILTIN account

### ISSUE THREAD

**Issue by michaelnoonan** [OctopusDeploy/Issues#2264]
_This same behaviour is likely anywhere *Custom User* is allowed as an option._

**Steps to Reproduce**

1. Create a Package Step and add the Windows Service feature
2. Select `LocalSystem` for the **Service Account**
3. Save, looking at the POST-ed JSON (and persisted document)
4. Now edit the Step selecting `Custom user...` for the **Service Account**, and enter a username and password
5. Save looking at the POST-ed JSON and the `CustomAccountName` and `CustomAccountPassword` properties are correctly added
6. Now edit the Step selecting `Network Service` for the **Service Account**
7. Save, looking at the POST-ed JSON (and persisted document) and the `CustomAccountName` and `CustomAccountPassword` properties are retained **even though they should have been cleared**.

The flow on effect of this is that offline drops are required to have a password for sensitive variables even though there aren't really supposed to be any sensitive variables.

**Configuring the Custom Account**

![image](https://cloud.githubusercontent.com/assets/1627582/12082303/573d9636-b2dc-11e5-87f6-f547a6a0d613.png)

**Switching back to the Network Service BUILTIN Account**

![image](https://cloud.githubusercontent.com/assets/1627582/12082309/6cb81a40-b2dc-11e5-96b3-a016d5803bae.png)

Source: http://help.octopusdeploy.com/discussions/problems/43593-offline-package-drop-incorrectly-thinks-there-are-sensitive-variables-and-requires-password

**Comment by alfhenrik** [comment]
Currently when we save a deployment process we restore any sensitive variables that an action previously had, so even if you change the user account to be used, the password will be added back. 

```
IEnumerable<KeyValuePair<string, PropertyValue>> WriteProperties(IEnumerable<KeyValuePair<string, string>> properties, IDictionary<string, string> sensitiveProperties, PropertiesDictionary existing)
        {
            //Create property collection from sensitive variables
            var result = sensitiveProperties
                .Where(p => !string.IsNullOrWhiteSpace(p.Value) 
                    && !string.IsNullOrWhiteSpace(p.Key))
                .ToDictionary(s => s.Key, s => new PropertyValue(s.Value, true));

            // Add any existing sensitive variables back onto the result if they havent been set
            existing?.Where(e => e.Value.IsSensitive
                && !result.ContainsKey(e.Key))
                .ToList()
                .ForEach(missingSensitive => result.Add(missingSensitive.Key, missingSensitive.Value));

            // Add Non sensitive values
            properties
                .Where(property => !result.ContainsKey(property.Key)
                    && !string.IsNullOrWhiteSpace(property.Value)
                    && !string.IsNullOrWhiteSpace(property.Key))
                .ToList()
                .ForEach(property => result.Add(property.Key, new PropertyValue(property.Value)));
            
            return result;
        }
```

Currently the only workaround is to remove the feature, save the process, and then add the feature back using one of the BUILTIN accounts.

This affects both the Windows Service and IIS Website features as they allow you to specify custom users for the Windows service and the IIS App Pool.

#2270 could be related to this issue as well.

Would probably be best to discuss this further within the team to come up with a fix.

**Comment by MJRichardson** [comment]
Release note:Windows Service Custom account and password properties no longer retained when switching account types.

**Comment by MJRichardson** [comment]
Resolved in Octopus 3.3.0
